subsys/bluetooth/host/scan.c

```diff
@@ -805,6 +805,14 @@ void bt_hci_le_per_adv_report(struct net_buf *buf)
 			 */
 			bt_hci_le_per_adv_report_recv(per_adv_sync, &buf->b, &info);
 		} else {
+			if (evt->data_status == BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_INCOMPLETE) {
+				LOG_DBG("Received incomplete advertising data. "
+					"Advertising report dropped.");
```

Comment:
I'm not sure exactly what it would be used for, it was something I considered when making the change, if the data should just be passed to the application as incomplete and reset the buffer or drop the data before resetting the before. 
I chose to drop the data as the same was already being done for the extended scanner: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/scan.c#L634-L640

Reply:
Ok. I have a feeling we might later have to add this back, possibly through an optional `Kconfig` option, in case an app for whatever reason wants to see even these packets.

Reply:
Anyway, I'm fine with making this consistent with the extended scanner for now.

Reply:
I think the reason is to not have to cache on controller side,  you may have up to 7 chains with ~2.4s spacing for each of scanned device

Reply:
I can create a separate issue/PR with a kconfig for this after its merged if that suits?

Reply:
I remember we discussed this in detail when it was added. 

We definitely wanted to provide reassembled data to the upper layers, so they did not have to implement reassembly themselves. 

I don't fully recall the argument to drop partial reports, rather than providing them to the upper layers. Perhaps the idea was that we wanted to avoid providing an advertising report where the length field of the report, did not match the actual report, as that would also mean that something `bt_data_parse` would not work with the received report.